### PR TITLE
Feature/mono eyrie

### PR DESCRIPTION
**Pull request by grg-haas**
Integrate the runtime into the monorepo

### CODE REVIEW

**Comment by dayeol** [comment]
@grg-haas is this ready for a review? Thanks!

**Comment by grg-haas** [comment]
Yes, it’s ready!

**Comment by grg-haas** [comment]
The previous force-push was just a rebase onto the latest master

**Comment by dayeol** [comment]
Hi, thank you so much for the PR.

nit: I think we should persist git history for eyrie runtime.
I referred to this article: https://medium.com/walkme-engineering/how-to-merge-a-git-submodule-into-its-main-repository-d83a215a319c

Other than that, I think the PR looks great. Do you mind to make the change?

**Comment by grg-haas** [comment]
@dayeol @anydayeol I believe this should be working now. I followed the advice on the medium article which you sent, but this caused some issues where git would get horrendously confused. Essentially, the article recommended adding a commit to the *end* of the submodule's history, which relocates everything into the target folder. However, this leads to some issues where git would believe that (for example), the submodule's `gitignore` is the same as the containing repository's `gitignore` until this move commit happens. This could lead to some serious issues with future rebases of the main `keystone` project. Therefore, I went through the `keystone-runtime` repository and modified commits such that each new file was directly introduced in the `runtime/` subdirectory. This means that the history reflected here is a lot cleaner and more self contained than the suggested approach. If you have worries about missing any changes, you can check out my corresponding branch in `keystone-runtime`: https://github.com/keystone-enclave/keystone-runtime/tree/move-files

File: sdk/macros.cmake [hunk 1]
```diff
@@ -95,11 +95,18 @@ macro(add_eyrie_runtime target_name tag plugins) # the files are passed via ${AR
...
+  else()
+    message(FATAL_ERROR "Don't know how to find runtime from current directory" ${SRCDIR})
+  endif()
+
   ExternalProject_Add(eyrie-${target_name}
     PREFIX ${runtime_prefix}
-    GIT_REPOSITORY https://github.com/keystone-enclave/keystone-runtime
-    GIT_TAG ${tag}
-    UPDATE_COMMAND git fetch
+    DOWNLOAD_COMMAND rm -rf ${eyrie_src} && cp -ar ${EYRIE_SRCDIR} ${eyrie_src}
```
Comment (dayeol) [inline sdk/macros.cmake:109]: Quick question: what happens if someone modifies the runtime source code for a certain binary by directly modifying the files in `${eyrie_src}`, and then run make? Will it remove all the changes and re-sync the directory with ${EYRIE_SRCDIR} ?

Comment (grg-haas) [reply sdk/macros.cmake:109]: From my experience running this code, this source copying only happens at cmake initialization time. Therefore, any changes in `${eyrie_src}` would persist through most of the lifetime of a user's build directory. Of course, the better way to develop a runtime change would be to modify files in `${EYRIE_SRCDIR}` directly :)

**Comment by grg-haas** [comment]
As a side note, I see that the default merge method for the keystone project is squash. For this PR, we should probably override this to ... rebase?

**Comment by dayeol** [comment]
I think we should temporarily enable merge commit, and then change it back to squash only :)
I think this is good to go, I'll announce it to folks before merging so that they don't get confused.

**Comment by grg-haas** [comment]
Sounds good to me! Feel free to take the lead on approving and merging this PR

**Pull request merged**